src/agni/server/dataguard/I_UserDataGuard.java

```diff
@@ -19,4 +19,16 @@
     public void changeUserCurrentChat(String user, String chatname);
 
     public String userCurrentChat(String user);
+
+	public boolean groupExists(String groupName);
+
+	public String getUsername(String ip);
+
+	public void addUserToChat(String username, String groupName);
+
+	public void removeUserFromChat(String username, String groupName);
```

Comment:
For the guards, anything to do with chats should be in the  `GroupChatDataGuard`. The `UserManager` can still use these functions, it still have access to both the `UserDataGuard` and the `GroupChatDataGuard`.